src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -1,138 +1,480 @@
-import { chakra } from "@chakra-ui/react";
-import React, { forwardRef } from "react";
-
-import MultiSelectListBox from "./MultiSelectListbox";
-import MultiSelectDialog from "./MultiSelectDialog";
+import React, { useState, forwardRef, useCallback, useRef } from "react";
+import { Box, chakra, useMultiStyleConfig } from "@chakra-ui/react";
+import Accordion from "./../Accordion/Accordion";
+import Button from "./../Button/Button";
+import CheckboxGroup from "./../CheckboxGroup/CheckboxGroup";
+import Checkbox from "./../Checkbox/Checkbox";
+import MultiSelectItemsCountButton from "./MultiSelectItemsCountButton";
+import TextInput from "../TextInput/TextInput";
 
 export interface MultiSelectItem {
   id: string;
   name: string;
+  isDisabled?: boolean;
   children?: MultiSelectItem[];
 }
-
-export const multiSelectWidthsArray = [
-  "default",
-  "fitContent",
-  "full",
-] as const;
+export const multiSelectWidthsArray = ["fitContent", "full"] as const;
 export type MultiSelectWidths = typeof multiSelectWidthsArray[number];
-
+export const multiSelectListOverflowArray = ["scroll", "expand"] as const;
+export type MultiSelectListOverflowTypes =
+  typeof multiSelectListOverflowArray[number];
 export interface SelectedItems {
   [name: string]: { items: string[] };
 }
-interface MultiSelectCommonProps {
-  /** The id of the MultiSelect. */
+type MultiSelectOnChange = (event: React.ChangeEvent<HTMLInputElement>) => void;
```

Comment:
Remove this custom type for the change handler function. It seems that in the DS the pattern is to use the `(event: React.ChangeEvent<HTMLInputElement>) => void` directly on the prop declaration.
This implementation caused an issue in the consuming app as I was unable to match the type. 
The other option would be to export this type and make it available for the consuming app but it seems that that would be a new pattern (see Button,  Checkbox, Select etc.)